### PR TITLE
Nowcast sensor endpoint

### PR DESCRIPTION
**Pull request by chinandrew**
Based off #355, only new commits are f3c84f2 and onward. Basically steals a bunch of the covidcast code for the PHP and uses a s lightly different table definition (which is still to be finalized).

TODO:
- [x] more integration tests, especially testing the acquisition flow.
- [x] update client

@krivard for review (let me know if theres anyone else I should tag)

cc: @mariajahja

### CODE REVIEW

**Review by krivard — APPROVED** [review]
lgtm

File: src/server/api.php [hunk 1]
```diff
@@ -2028,6 +2092,44 @@ function meta_delphi() {
...
+        foreach ($epidata as $row) {
+          if ($key != $row['signal']) {
+            $key = $row['signal'];
+            $epi_tree[$key] = array();
+          }
+          unset($row['signal']);
+          array_push($epi_tree[$key],$row);
+        }
+        $epidata = array($epi_tree);
+      }
```
Comment (krivard) [inline src/server/api.php:2130]: recommend dropping this unless you have an actual use case for multi-signal queries with results grouped by signal

**Review by krivard — COMMENTED** [review]
Whoops, github lied to me about how many changes were in here. Not done yet, I'll be able to take a closer look tomorrow.

**Comment by chinandrew** [comment]
> Whoops, github lied to me about how many changes were in here. Not done yet, I'll be able to take a closer look tomorrow.

No worries, it's certainly on the longer side. Let me know if I can make the commits more clear (the first one, f3c84f2, is the big one.)

File: src/server/api.php (hunk 1 above)
Comment (chinandrew) [reply src/server/api.php:2130]: good call, will remove.

File: integrations/client/test_delphi_epidata.py [hunk 2]
```diff
@@ -385,3 +386,88 @@ def test_covidcast_meta(self):
...
+        'geo_value': '01001',
+        'value': 1.5,
+        'issue': 20200102,
+        'lag': 2,
+        'signal': 'sig1',
+      }],
+      'message': 'success',
+    })
+
+    # request as_od
```
Comment (krivard) [inline integrations/client/test_delphi_epidata.py:451]:
```suggestion
    # request as_of
```

File: integrations/server/test_covidcast_nowcast.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    self.cnx = cnx
+    self.cur = cnx.cursor()
+
+  def tearDown(self):
+    """Perform per-test teardown."""
+    self.cur.close()
+    self.cnx.close()
+
+  def test_query_without(self):
+    """Query without specifying an issue."""
```
Comment (krivard) [inline integrations/server/test_covidcast_nowcast.py:42]:
```suggestion
  def test_query(self):
    """Query nowcasts using default and specified issue."""
```

File: integrations/server/test_covidcast_nowcast.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      'epidata': [{
+        'signal': 'sig',
+        'time_value': 20200101,
+        'geo_value': '01001',
+        'value': 3.5,
+        'issue': 20200101,
+        'lag': 2,
+       }],
+      'message': 'success',
+    })
```
Comment (krivard) [inline integrations/server/test_covidcast_nowcast.py:127]: newline?

File: src/client/delphi_epidata.py [hunk 5]
```diff
@@ -665,3 +665,44 @@ def covid_hosp_facility_lookup(
       raise Exception('one of `state`, `ccn`, `city`, `zip`, or `fips_code` is required')
     # Make the API call
     return Epidata._request(params)
+
+  # Fetch Delphi's COVID-19 Surveillance Streams
```
Comment (krivard) [inline src/client/delphi_epidata.py:669]: No idea why we comment these twice, but they should both be correct

```suggestion
  # Fetch Delphi's COVID-19 Nowcast sensors
```

File: src/client/delphi_epidata.py [hunk 6]
```diff
@@ -665,3 +665,44 @@ def covid_hosp_facility_lookup(
...
+  @staticmethod
+  def covidcast_nowcast(
+          data_source, signals, time_type, geo_type,
+          time_values, geo_value, as_of=None, issues=None, lag=None, **kwargs):
+    """Fetch Delphi's COVID-19 Nowcast sensors"""
+    # also support old parameter name
+    if signals is None and 'signal' in kwargs:
+      signals=kwargs['signal']
```
Comment (krivard) [inline src/client/delphi_epidata.py:677]: This is a new endpoint so there is no need to make it backwards-compatible with anything
```suggestion
```

File: src/client/delphi_epidata.py [hunk 7]
```diff
@@ -665,3 +665,44 @@ def covid_hosp_facility_lookup(
...
+    if issues is not None:
+      params['issues'] = Epidata._list(issues)
+    if lag is not None:
+      params['lag'] = lag
+
+    if 'format' in kwargs:
+      params['format'] = kwargs['format']
+
+    # Make the API call
+    return Epidata._request(params)
```
Comment (krivard) [inline src/client/delphi_epidata.py:708]: newline?

File: src/server/api.php [hunk 8]
```diff
@@ -2028,6 +2092,30 @@ function meta_delphi() {
       $epidata = get_covid_hosp_facility_lookup($state, $ccn, $city, $zip, $fips_code);
       store_result($data, $epidata);
     }
+  } else if($source === 'covidcast_nowcast') {
+    if(require_all($data, array('data_source', 'time_type', 'geo_type', 'time_values'))
+       && require_any($data, array('signal', 'signals'))
```
Comment (krivard) [inline src/server/api.php:2097]: i mean I guess you could do this as accepting either `signal` or `signals` just to be nice, but it's up to you, there's arguably no existing user base to consider
```suggestion
    if(require_all($data, array('data_source', 'time_type', 'geo_type', 'time_values', 'signals'))
```

**Review by krivard — CHANGES_REQUESTED** [review]
Some docs and naming to fix. optionally nuke the backwards-compatible parameter names. if you'd rather maintain parallel syntax with covidcast, then annotate both copies to that effect, so we don't forget.

**Comment by chinandrew** [comment]
Thanks for catching. Removed the old `signal` arguments

**Review by krivard — APPROVED** [review]
nice work

**Pull request merged**